### ISSUE TITLE
FTP Driver error when calling the exist function

### ISSUE THREAD

**Issue by pmartelletti** [thephpleague/flysystem#554]
When I try to see if a file exists (or not) using the FTP Adapter, and calling the following function:

`$ftp->exists($destination)`

some FTP servers return the following output to `ftp_rawlist`:

> destination_file not found

Instead of an empty string or array. And therefore, this (on the `getMetadata` function of the FTP adapter):

```
$listing = ftp_rawlist($connection, $path);
if (empty($listing)) {
    return false;
}

return  $this->normalizeObject($listing[0], '');
```

is not returning false, even though the file does not exists. Therefore, is trying to normalize an object that does not have the correct input.

Does this makes sense to you guys?

**Comment by frankdejonge** [comment]
@pmartelletti have you got more info on what type of ftp server you're talking to?

**Comment by pmartelletti** [comment]
@frankdejonge not right now, but I can ask the customer to give more feedback on this. What info would be usefull?

**Comment by frankdejonge** [comment]
@pmartelletti some googling lead me to believe it might be a windows ftp server. Any info would be nice to get a better look at the situation.

**Comment by pmartelletti** [comment]
@frankdejonge sure. I'll get back to you with further details about the FTP server, but it seems to be that we would need to do some system detection before the normalization of the object as well. :)

**Comment by pmartelletti** [comment]
@frankdejonge the FTP server is a private warehouse so I could not get any more details on this, but I've created a PR that would, in theory, solve this. See #555. :smile:

**Comment by frankdejonge** [comment]
This issue was fixed by #555
